indexer/map_object.cpp

```diff
@@ -94,6 +94,37 @@ string MapObject::GetLocalizedType() const
   return platform::GetLocalizedTypeName(classif().GetReadableObjectName(copy.GetBestType()));
 }
 
+std::string MapObject::GetAllLocalizedTypes() const
+{
+  ASSERT(!m_types.Empty(), ());
+  feature::TypesHolder copy(m_types);
+  copy.SortBySpec();
+
+  auto const & isPoi = ftypes::IsPoiChecker::Instance();
+  auto const & isGeneralAmenity = ftypes::IsGeneralAmenityChecker::Instance();
+
+  std::ostringstream oss;
+  bool first = true;
+  for (auto const type : copy)
+  {
+    // Ignore types that are not POI
+    // Ignore general "amenity" types that have no subtypes
+    // But always show the main type
+    if ((!isPoi(type) || isGeneralAmenity(type)) && !first)
```

Comment:
Will it work properly by leaving "amenity" where it's the only one type of a feature, and where types are "amenity" + "something" or "something" + "amenity" ?

Reply:
The first type is always shown, and amenity is ignored if it's not the main type so:
"amenity" shows "Amenity"
"amenity"+"something" shows "Amenity  •  Something"
"something"+"amenity" shows "Something"